app/controllers/newsletter_signups_controller.rb

```diff
@@ -0,0 +1,30 @@
+class NewsletterSignupsController < ApplicationController
+  before_action :set_newsletter_signup
+
+  def create
+    response = request_signup
+    if response
+      redirect_to(root_path, success: 'Thank you for signing up!')
+    else
+      redirect_to(root_path, error: 'There was an error with your request, please try again.')
+    end
```

Comment:
I'm not sure I'm getting the error handling exactly right, looking forward to suggestions!